pkg/gds/models/v1/models.go

```diff
@@ -45,6 +45,51 @@ func SetAdminVerificationToken(vasp *pb.VASP, token string) (err error) {
 	return nil
 }
 
+// GetVerificationChangeLog from the extra data on the VASP record.
+func GetVerificationChangeLog(vasp *pb.VASP) (_ []*VerificationChangeLogEntry, err error) {
+	// If the extra data is nil, return nil (no change log).
+	if vasp.Extra == nil {
+		return nil, nil
+	}
+
+	// Unmarshal the extra data field on the VASP.
+	extra := &GDSExtraData{}
+	if err = vasp.Extra.UnmarshalTo(extra); err != nil {
+		return nil, err
+	}
+	return extra.GetVerificationChangeLog(), nil
+}
+
+// Append a VerificationChangeLogEntry to the extra data on the VASP record.
+func AppendVerificationChangeLog(vasp *pb.VASP, entry *VerificationChangeLogEntry) (err error) {
+	// Validate current state.
+	if entry.CurrentState > uint32(pb.VerificationState_ERRORED) {
+		return fmt.Errorf("cannot set verification state to unsupported value %d", entry.CurrentState)
+	}
+
+	// Unmarshal previous extra data.
+	extra := &GDSExtraData{}
+	if vasp.Extra != nil {
+		if err = vasp.Extra.UnmarshalTo(extra); err != nil {
+			return fmt.Errorf("could not deserialize previous extra: %s", err)
+		}
+	}
```

Comment:
If `vasp.Extra == nil` you'll need to `make` the array of audit logs; e.g. 

```go
} else {
    extra.VerificationChangeLog = make([]*VerificationChangeLogEntry, 0, 1) 
}
```

I've made the slice with a capacity of 1 since you know you'll be appending something shortly.